src/documents/consumer.py

```diff
@@ -63,59 +64,124 @@ def run(self) -> Optional[str]:
+                        if action.assign_title is not None:
+                            action_overrides.title = action.assign_title
+                        if action.assign_tags is not None:
+                            action_overrides.tag_ids = [
+                                tag.pk for tag in action.assign_tags.all()
+                            ]
+                        if action.assign_correspondent is not None:
+                            action_overrides.correspondent_id = (
+                                action.assign_correspondent.pk
+                            )
+                        if action.assign_document_type is not None:
+                            action_overrides.document_type_id = (
+                                action.assign_document_type.pk
+                            )
+                        if action.assign_storage_path is not None:
+                            action_overrides.storage_path_id = (
+                                action.assign_storage_path.pk
+                            )
+                        if action.assign_owner is not None:
+                            action_overrides.owner_id = action.assign_owner.pk
+                        if action.assign_view_users is not None:
+                            action_overrides.view_users = [
+                                user.pk for user in action.assign_view_users.all()
+                            ]
+                        if action.assign_view_groups is not None:
+                            action_overrides.view_groups = [
+                                group.pk for group in action.assign_view_groups.all()
+                            ]
+                        if action.assign_change_users is not None:
+                            action_overrides.change_users = [
+                                user.pk for user in action.assign_change_users.all()
+                            ]
+                        if action.assign_change_groups is not None:
+                            action_overrides.change_groups = [
+                                group.pk for group in action.assign_change_groups.all()
+                            ]
+                        if action.assign_custom_fields is not None:
+                            action_overrides.custom_field_ids = [
+                                field.pk for field in action.assign_custom_fields.all()
+                            ]
+                        overrides.update(action_overrides)
+                    elif action.type == WorkflowAction.WorkflowActionType.REMOVAL:
+                        # Removal actions overwrite the current overrides
+                        if action.remove_all_tags:
+                            overrides.tag_ids = []
+                        elif action.remove_tags.all().count() > 0:
+                            for tag in action.remove_tags.all():
+                                overrides.tag_ids.remove(tag.pk)
```

Comment:
Does somewhere ensure that the removed item will be in the list?  Removal if it doesn't exist will fail.

This may apply to custom field IDs too, anything removing an item from the list